File: leetcode/0009. Palindrome Number/Solution.swift
// https://leetcode.com/problems/palindrome-number/

class Solution {
    func isPalindrome(_ x: Int) -> Bool {
        if x < 0 {
            return false
        }

        if x > 9 && x.isMultiple(of: 10) {
            return false
        }

        let original = x
        var flipped = 0
        var x = x
        while x > 0 {
            flipped = 10 * flipped + x % 10
            x /= 10
        }

        return original == flipped
    }
}

File: leetcode/0009. Palindrome Number/Solution.py
# https://leetcode.com/problems/palindrome-number/

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        if x > 9 and not x % 10:
            return False

        original = x
        flipped = 0
        while x:
            flipped = 10 * flipped + x % 10
            x //= 10

        return original == flipped

s = Solution()
assert s.isPalindrome(121) == True
assert s.isPalindrome(-121) == False
assert s.isPalindrome(10) == False